cmd/autometrics/main.go

```diff
@@ -1,51 +1,77 @@
 package main
 
 import (
+	"fmt"
 	"log"
-	"os"
+	"strings"
 
 	internal "github.com/autometrics-dev/autometrics-go/internal/autometrics"
+	"github.com/autometrics-dev/autometrics-go/internal/build"
 	"github.com/autometrics-dev/autometrics-go/internal/generate"
 	"github.com/autometrics-dev/autometrics-go/pkg/autometrics"
+
+	arg "github.com/alexflint/go-arg"
 )
 
 const (
-	prometheusAddressEnvironmentVariable = "AM_PROMETHEUS_URL"
-	useOtelFlag                          = "-otel"
-	allowCustomLatencies                 = "-custom-latency"
-	DefaultPrometheusInstanceUrl         = "http://localhost:9090/"
+	DefaultPrometheusInstanceUrl = "http://localhost:9090/"
 )
 
-func main() {
-	fileName := os.Getenv("GOFILE")
-	moduleName := os.Getenv("GOPACKAGE")
-	args := os.Args
+type args struct {
+	FileName             string `arg:"-f,--,required,env:GOFILE" placeholder:"FILE_NAME" help:"File to transform."`
+	ModuleName           string `arg:"-m,--,required,env:GOPACKAGE" placeholder:"MODULE_NAME" help:"Module containing the file to transform."`
+	PrometheusUrl        string `arg:"--prom_url,env:AM_PROMETHEUS_URL" placeholder:"PROMETHEUS_URL" default:"http://localhost:9090" help:"Base URL of the Prometheus instance to generate links to."`
+	UseOtel              bool   `arg:"--otel" default:"false" help:"Use OpenTelemetry client library to instrument code instead of default Prometheus."`
```

Comment:
Doesn't have to be in this PR but I think we should change the default to be OTel for all the libraries